### PR TITLE
chore: run tests on PR

### PR DESCRIPTION
**Pull request by eliobischof**
- Removes test flakiness
- Runs tests on PR
- Adds a PR template

### CODE REVIEW

**Review by muhlemmer — CHANGES_REQUESTED** [review]
When following the steps from `CONTRIBUTING.md` locally, migrations fail at step 3.

```
docker compose --file ./acceptance/docker-compose.yaml run wait_for_zitadel
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
[+] Running 3/0
 ⠿ Network acceptance_default      Created                                                                                                 0.0s
 ⠿ Container acceptance-db-1       Created                                                                                                 0.0s
 ⠿ Container acceptance-zitadel-1  Created                                                                                                 0.0s
[+] Running 2/2
 ⠿ Container acceptance-db-1       Healthy                                                                                                10.8s
 ⠿ Container acceptance-zitadel-1  Started                                                                                                 0.3s
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadeldocker compose --file ./acceptance/docker-compose.yaml run wait_for_zitadel
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
[+] Running 3/0
 ⠿ Network acceptance_default      Created                                                                                                 0.0s
 ⠿ Container acceptance-db-1       Created                                                                                                 0.0s
 ⠿ Container acceptance-zitadel-1  Created                                                                                                 0.0s
[+] Running 2/2
 ⠿ Container acceptance-db-1       Healthy                                                                                                10.8s
 ⠿ Container acceptance-zitadel-1  Started                                                                                                 0.3s
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (7) Failed to connect to zitadel port 8080 after 0 ms: Couldn't connect to server
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
curl: (6) Could not resolve host: zitadel
^C
curl: (6) Could not resolve host: zitadel
^C
```
Logs:

```
timz@ztdl ~/zitadel/terraform-provider-zitadel $ docker compose --file ./acceptance/docker-compose.yaml logs zitadel
WARN[0000] The "UID" variable is not set. Defaulting to a blank string. 
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="initialization started" caller="/home/runner/work/zitadel/zitadel/cmd/initialise/init.go:72"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify user" caller="/home/runner/work/zitadel/zitadel/cmd/initialise/verify_user.go:38" username=zitadel
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify database" caller="/home/runner/work/zitadel/zitadel/cmd/initialise/verify_database.go:38" database=zitadel
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify grant" caller="/home/runner/work/zitadel/zitadel/cmd/initialise/verify_grant.go:33" database=zitadel user=zitadel
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify zitadel" caller="/home/runner/work/zitadel/zitadel/cmd/initialise/verify_zitadel.go:69" database=zitadel
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="setup started" caller="/home/runner/work/zitadel/zitadel/cmd/setup/setup.go:60"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify migration 01_tables" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="starting migration 01_tables" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:49"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify migration 02_assets" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="starting migration 02_assets" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:49"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="verify migration 03_default_instance" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:39"
acceptance-zitadel-1  | time="2023-05-23T13:06:35Z" level=info msg="starting migration 03_default_instance" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:49"
acceptance-zitadel-1  | time="2023-05-23T13:06:37Z" level=error msg="migration failed" caller="/home/runner/work/zitadel/zitadel/internal/migration/migration.go:51" error="open /machinekey/zitadel-admin-sa.json: permission denied"
acceptance-zitadel-1  | time="2023-05-23T13:06:37Z" level=fatal msg="unable to migrate step 3" caller="/home/runner/work/zitadel/zitadel/cmd/setup/setup.go:109" error="open /machinekey/zitadel-admin-sa.json: permission denied"
```

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -6,26 +6,28 @@
...
 
 # Run Acceptance Tests
 
-Ensure ZITADEL listens at http://localhost:8080 and you have a service account key in your local filesystem.
-The easiest way to achieve that is [to follow this guide](https://zitadel.com/docs/self-hosting/deploy/compose#docker-compose-with-service-account).
+Run a local ZITADEL instance using docker compose.
 
 ```bash
-TF_ACC=1 TF_ACC_ZITADEL_TOKEN=/my-token.json go test ./...
+docker compose --file ./acceptance/docker-compose.yaml run wait_for_zitadel
```
[the remark this answers is not in the document]
Comment (eliobischof) [reply CONTRIBUTING.md:23]: what docker version do you have?

[the remark this answers is not in the document]
Comment (eliobischof) [reply CONTRIBUTING.md:30]: I an explicit export to the contributing guide https://github.com/zitadel/terraform-provider-zitadel/pull/97/commits/78de7825f98f05d12c9a6f4f1a79abfbfb56b729

**Comment by eliobischof** [comment]
@muhlemmer I added the retry https://github.com/zitadel/terraform-provider-zitadel/pull/97/commits/2e8e13f854051538ad734d348d1c0d79f762f876 and the docker pull https://github.com/zitadel/terraform-provider-zitadel/pull/97/commits/93d6c900b1adb48097f6a1dac532f52780f65dfd

The tests succeed reliably on my machine:
```bash
while go test ./...; do go clean -testcache; done
```

**Comment by eliobischof** [comment]
I'm still suspicious, will add a unit test

**Review by muhlemmer — APPROVED** [review]
all good now :)

**Pull request merged**